### PR TITLE
Marketplace api openapi spec and client

### PR DESCRIPTION
**Pull request by ohmpatel1997**
This PR adds the openapi spec for the public marketplace apis. we can generate client from the spec as well

### CODE REVIEW

**Review by FSM1 — APPROVED** [review]
lgtm 🎉

File: docs/marketplace-api/nswag/marketplace-api-client.nswag [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
...ress\"\n          },\n          \"buyer\": {\n            \"type\": \"string\",\n            \"description\": \"Buyer address\"\n          },\n          \"price\": {\n            \"type\": \"string\",\n            \"description\": \"Price of the item\"\n          },\n          \"payment_token\": {\n            \"type\": \"string\",\n            \"description\": \"Payment token\"\n          },\n          \"listed_at\": {\n            \"type\": \"integer\",\n            \"description\": \"Listed at epoch time\"\n          },\n          \"status\": {\n            \"$ref\": \"#/components/schemas/item_status\"\n          },\n          \"page_number\": {\n            \"type\": \"integer\",\n            \"description\": \"Page number\"\n          },\n          \"page_size\": {\n            \"type\": \"integer\",\n            \"description\": \"Page size\"\n          },\n          \"total\": {\n            \"type\": \"integer\",\n            \"description\": \"Total number of items\"\n          },\n          \"cursor\": {\n            \"type\": \"string\",\n            \"description\": \"Pagination cursor, if empty no more items to return\"\n          }\n        }\n      }\n    }\n  }\n}"
```
[the remark this answers is not in the document]
Comment (FSM1) [reply docs/marketplace-api/nswag/marketplace-api-client.nswag]: This is now resolved

**Comment by sneakzttv** [comment]
Awaiting conversations with Jay and Arun RE marketplace, will modify and merge after.

File: docs/marketplace-api/spec/components/item_response.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+type: object
+properties:
+  id:
+      type: string
+      description: Item ID
+  token:
```
Comment (dhyaniarun1993) [inline docs/marketplace-api/spec/components/item_response.yaml:6]: The spec is incorrect here.

File: docs/marketplace-api/spec/components/item_status.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+type: string
+enum:
+  - invalid
```
Comment (dhyaniarun1993) [inline docs/marketplace-api/spec/components/item_status.yaml:3]: We should remove this

File: docs/marketplace-api/spec/components/item_response.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  seller:
+      type: string
+      description: Seller address
+  buyer:
+      type: string
+      description: Buyer address
+  price:
+      type: string
+      description: Price of the item
+  payment_token:
```
Comment (dhyaniarun1993) [inline docs/marketplace-api/spec/components/item_response.yaml:21]: This isn't part of the spec

**Pull request closed without merging**